### DISCUSSION TITLE
Hello team how to use the local model files in language_classification

### DISCUSSION THREAD

**Discussion by TapendraBaduwal** [PaddlePaddle/PaddleOCR#7735] [General]
Hello team how to use the local model files in language_classification

    from paddleclas import PaddleClas
    clas = PaddleClas(self.lang_classification_model_path, self.lang_classification_dir_path)
    infer_imgs = '/home/tapendra/Desktop/kapediaml/images/docnp (1).png'
    result=clas.predict(infer_imgs)
    print(next(result))

error:If you want to use your own model, please specify inference_model_dir!

**Comment by TingquanGao** [comment]
Hi，please specify the directory that should contain the inference.pdmodel and inference.pdiparams files through parameter `inference_model_dir`, such as:

```python
clas = PaddleClas(inference_model_dir='./inference/')
```
In the above code, there must be  inference.pdmodel and inference.pdiparams files in the directory './inference/'.

**Reply by TapendraBaduwal** [reply]
@TingquanGao 
These three files from .paddlecls/inference_model/PULC/language_classification/
Model download from https://github.com/PaddlePaddle/PaddleClas/blob/develop/docs/en/PULC/PULC_model_list_en.md

inference.pdiparams
inference.pdiparams.info
inference.pdmodel

 Using local model then output is different classes...i need right model
[{'class_ids': [9, 1, 4, 6, 3], 'scores': [0.66337, 0.12631, 0.08265, 0.02806, 0.02192], 'label_names': ['ostrich, Struthio camelus', 'goldfish, Carassius auratus', 'hammerhead, hammerhead shark', 'stingray', 'tiger shark, Galeocerdo cuvieri']}]
[2022/09/29 16:45:24] ppcls WARNING: The current running environment does not support the use of GPU. CPU has been used instead.

**Reply by TingquanGao** [reply]
That is because the PaddleClas whl use the class id map and preprocesses of ImageNet1k by default when local inference files used. And it is not be supported that specify the related inference configuration now. We are considering supportting this function. For the problem, you could try to move the inference.pdmodel and inference.pdiparams files to the directory `~/.paddleclas/inference_model/PULC/language_classification/` as the temporary solution.

**Reply by TapendraBaduwal** [reply]
@TingquanGao 

As per your view i put only two file in  the directory inference.pdmodel and inference.pdiparams .
Even not working please look at once in my image as below
![Screenshot from 2022-09-29 20-17-58](https://user-images.githubusercontent.com/87867535/193061634-b963fe90-e04b-4df4-91ad-8d19d1345f94.png)
